db.php

```diff
@@ -773,7 +779,12 @@ public function db_connect( $query = '' ) {
 
 		$this->set_charset( $this->dbhs[ $dbhname ], $charset, $collate );
 
-		$this->dbh = $this->dbhs[ $dbhname ]; // needed by $wpdb->_real_escape()
+		$this->dbh = $this->dbhs[ $dbhname ]; // needed by $wpdb->_real_escape() and `$wpdb->set_sql_mode()`
+
+		// Set the SQL mode 
+		if ( is_callable( $this, 'set_sql_mode' ) ) {
```

Comment:
```suggestion
		if ( is_callable( array($this, 'set_sql_mode') ) ) {
```

With this small modification it works using WP 6.1.1 with MySQL 8.0 🚀

Reply:
Thanks @viktorhaid - That was a silly on my part! The syntax was 100% incorrect :D